### PR TITLE
Feat/add haros to ci

### PR DESCRIPTION
**Pull request by stlaperriere**
Added Haros to the CI. [Haros](https://github.com/git-afsantos/haros) is a static analysis tool specifically made for ROS.

The results of the analysis will be published [here](https://swarmus.github.io/Haros-viz/#packages).

Configuration is still very minimal, there are a lot of unclear/undocumented features.

### CODE REVIEW

**Review by cquesnel — APPROVED** [review]

File: .github/workflows/haros.yml
Comment (xgroleau) [inline .github/workflows/haros.yml:35]: Why are we not using pip from the ubuntu repository?

File: .github/workflows/haros.yml
Comment (xgroleau) [inline .github/workflows/haros.yml:56]: I'm just curious, why on an other repository and not just another branch like the other? And what happens if we have multiple repository with the need to analayze ROS packages? Do they all publish to this repo?

File: .github/workflows/haros.yml
Comment (xgroleau) [inline .github/workflows/haros.yml:7]: Should we add on pull request on master/push (even if they will not be that common)? I mean we don't even pay for the runners anyway.

A bit out of scope, why do we have develop and master?

File: haros_config.yaml
Comment (xgroleau) [inline haros_config.yaml:3]: Why did this file already existed in develop? was it only to test the CI?

**Review by xgroleau — CHANGES_REQUESTED** [review]
Nice job, good CI is always appreciated! 👍 A couple question and inqueries about haros more than change requests. I have absolutely no knowledge of Haros. 

Should probably add something on the README about haros, the link to the github pages of haros-viz, etc.

**Comment by xgroleau** [comment]
Just saw something on haros-viz, why do we have 5.3k lines of code? the dependencies?

File: .github/workflows/haros.yml
Comment (stlaperriere) [reply .github/workflows/haros.yml:35]: Getting the right package to install pip/pip2 with python3/python2 has always been a little messy for me. Sometimes the `pip` command refers to `pip3` really, or `pip2`, and so on. This installer just makes sure that there is no conflict between pip2 and pip3.

File: .github/workflows/haros.yml
Comment (stlaperriere) [reply .github/workflows/haros.yml:56]: I didn't look so much into that, but I thought that it was impossible to host two seperate gh pages sites on a single repo. Am I wrong?

File: .github/workflows/haros.yml
Comment (stlaperriere) [reply .github/workflows/haros.yml:7]: Having `develop` is simply an old habit of mine. It feels easier having a develop branch to QA the integration of all the features before a release. But I wouldn't mind changing it.

I'll add triggers for branch `master`.

File: haros_config.yaml
Comment (stlaperriere) [reply haros_config.yaml:3]: Yeah i comitted this file by mistake in my last PR. Sowwyyyyy :man_shrugging:

**Comment by stlaperriere** [comment]
> Just saw something on haros-viz, why do we have 5.3k lines of code? the dependencies?

@xgroleau I would think so. Actually, with no HiveMind, there is only a few dozens/hundreds of lines.

File: .github/workflows/haros.yml
Comment (xgroleau) [reply .github/workflows/haros.yml:35]: Ok and I just checked an pip isn't even included on LST 20.04.  So that clears it up!

File: .github/workflows/haros.yml
Comment (xgroleau) [reply .github/workflows/haros.yml:56]: You are right we can't generate multiple github pages, though could we all put the doc together? as of now we have 3 modules, thus 3 link in your index.html in the doc github pages, can it be possible to add a fourth and it would be haros?

 How is the index.html from the docs ci created? If it's form the github action, probably it's possible to have them all on the same gh page

Comment (stlaperriere) [reply .github/workflows/haros.yml:56]: The index.html is generated by one of the python scripts, which is run from github actions. Thing is, the docs and Haros are run from two separate CI pipelines.
I could just create a root index.html and deploy the docs and haros stuff to two separate folders. I wanted to avoid messing with too many html pages, manually though, this is why I did it this way.

Comment (xgroleau) [reply .github/workflows/haros.yml:56]: Can't one CI pipeline generate all the doc? I would definitly prefer all the doc on the github pages but if it's too much hassle maybe just rename the haros doc repo for SwarmUS-ROS-Haros-viz so we know it's associated with it and just in case we have another use for Haros and want to visualise it. 

I'll approve the PR so you can merge once you've decided and done the changes

**Review by xgroleau — APPROVED** [review]

**Review by xgroleau — APPROVED** [review]
Approved, but check the comment on the discussion before merge

File: .github/workflows/haros.yml
Comment (stlaperriere) [reply .github/workflows/haros.yml:56]: Yeah right now the two CI pipelines do not have the same trigger conditions. I'll just rename the other repo.

**Pull request merged**
